hazelcast/src/main/java/com/hazelcast/core/Immutable.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.core;
+
+import com.hazelcast.spi.annotation.Beta;
+
+/**
+ * Allows notifying Hazelcast code that the object implementing this interface is effectively immutable.
```

Comment:
If we keep this text, it's `optimizations` (American English) and `Hazelcast` (uppercase).

Alternative:
```java
/**
 * Marks classes that are (effectively) immutable.
 * <p>
 * Instances of immutable classes either have no state (e.g. pure functions) or the state
 * is not mutated after the object construction (e.g. all fields are final). This allows
 * Hazelcast to apply performance optimizations such as avoiding cloning user supplied
 * objects or cloning Hazelcast internal objects supplied to the user.
 * <p>
 * It is important that to follows these rules:
 * <ul>
 * <li>the object must not have any state which is changed by cloning the object</li>
 * <li>the existing state must not be changed</li>
 * </ul>
 * <b>Warning:</b> If a class implements this interface but does not follow these rules,
 * the results of the execution are undefined.
 */
```